include/sound/sof/channel_map.h

```diff
@@ -0,0 +1,65 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Marcin Maka <marcin.maka@linux.intel.com>
+ *         Slawomir Blauciak <slawomir.blauciak@linux.intel.com>
+ */
+
+#ifndef __IPC_CHANNEL_MAP_H__
+#define __IPC_CHANNEL_MAP_H__
+
+#include <ipc/header.h>
+#include <stdint.h>
+
+/**
+ * \brief Channel map, specifies transformation of one-to-many or many-to-one.
+ *
+ * In case of one-to-many specifies how the output channels are computed out of
+ * a single source channel,
+ * in case of many-to-one specifies how a single target channel is computed
+ * from a multichannel input stream.
+ *
+ * Channel index specifies position of the channel in the stream on the 'one'
+ * side.
+ *
+ * Ext ID is the identifier of external part of the transformation. Depending
+ * on the context, it may be pipeline ID, dai ID, ...
+ *
+ * Channel mask describes which channels are taken into account on the "many"
+ * side. Bit[i] set to 1 means that i-th channel is used for computation
+ * (either as source or as a target).
+ *
+ * Channel mask is followed by array of coefficients in Q2.30 format,
+ * one per each channel set in the mask (left to right, LS bit set in the
+ * mask corresponds to ch_coeffs[0]).
+ */
+struct sof_ipc_channel_map {
+	uint32_t ch_index;
+	uint32_t ext_id;
+	uint32_t ch_mask;
+	int32_t ch_coeffs[0];
+} __packed;
+
+/**
+ * \brief Complete map for each channel of a multichannel stream.
+ *
+ * num_ch_map Specifies number of items in the ch_map.
+ * More than one transformation per a single channel is allowed (in case
+ * multiple external entities are transformed).
+ * A channel may be skipped in the transformation list, then it is filled
+ * with 0's by the transformation function.
+ */
+struct sof_ipc_stream_map {
+	struct sof_ipc_cmd_hdr hdr;
+	uint32_t num_ch_map;
+	struct sof_ipc_channel_map ch_map[0];
+} __packed;
+
+/** Maximum number of unique physical links in one-to-many configuration */
+#define CHANNEL_MAP_MAX_LINKS	4
+
+/** Maximum number for channel transformations in the map */
+#define CHANNEL_MAP_MAX_ELEMS	8
```

Comment:
you've got to explain what MAX_LINKS and MAX_ELEMS are.

Reply:
I've cleared up the comments for those a bit. Is it better?

Reply:
not really, no.
The MAX_LINKS talks about physical links, but the ipc_channel_map refers to an ext_id which can be a pipeline.
And the definition for the MAX_ELEMS talks about a stream which is not a clear definition.

I realize I am being thick but you guys need to use definitions that people without 5 years of Intel context can understand. I am not asking for the concepts to be changed, just that their definitions are clear and non-ambiguous. For example the notion of stream has a very specific meaning if I put my SoundWire hat on.

Reply:
After rethinking, I've come to a conclusion that those defines are not entirely necessary. Having that in mind, to avoid confusion I've decided to remove them.